### PR TITLE
Space out header

### PR DESCRIPTION
**Pull request by algomaster99**
### Header
Screenshot from 2019-07-07 17-30-02

- `onSelectNewVegaLite` is being repeated in file `src/components/input-panel/spec-editor/renderer.tsx` and `src/components/header/renderer.tsx`.
- This is marked `WIP` since the header requires some minor CSS fixes to improve spacing between the buttons.

### CODE REVIEW

**Comment by algomaster99** [comment]
@domoritz 
css
How are things looking?

**Comment by algomaster99** [comment]
@domoritz I am quite fine with adding the **Share URL** button one level deep inside export. But I am open to changes.

**Comment by domoritz** [comment]
> @domoritz I am quite fine with adding the Share URL button one level deep inside export. But I am open to changes.

We had a lengthy discussion about this on Slack. It would be better to have it at the top level since `share` isn't part of `export`.

**Comment by algomaster99** [comment]
@domoritz 
Screenshot from 2019-07-07 23-43-01
Share button has been restored.

**Comment by algomaster99** [comment]
@domoritz Shall we merge this?

**Comment by domoritz** [comment]
* [ ] Add back GIST loading
* [ ] Make format work (should use compact format)
* [ ] Update the help docs. We don't need to suggest F1 anymore. We should instead explain some common functionalities like formatting, clearing, ...
* [ ] Rename "Options" to "Commands"

**Comment by domoritz** [comment]
@siddhant1 We also need the help dialog cleanup for these changes (e.g. we want to docs links to be at the top of the help dialog).

**Comment by algomaster99** [comment]
@domoritz Format works if you run it through the command palette.

**Comment by domoritz** [comment]
Not for me:

editor

**Comment by algomaster99** [comment]
@domoritz It initially too worked like this.
format

I think it is a problem with the Monaco editor's formatter.

**Comment by algomaster99** [comment]
@domoritz I am thinking to modify it like this:
Screenshot from 2019-07-08 23-57-55

**Comment by domoritz** [comment]
> I think it is a problem with the Monaco editor's formatter.

Yep: https://github.com/microsoft/monaco-editor/issues/889

**Comment by algomaster99** [comment]
@domoritz Check the help modal too

**Comment by domoritz** [comment]
Looks good. Let's move the docs links up.

**Comment by algomaster99** [comment]
@domoritz Done!

File: src/components/header/help-modal/index.tsx [hunk 1]
```diff
@@ -31,9 +31,7 @@ const HelpModal = () => {
       <h1 className="modal-header">Help</h1>
       <h2>Keyboard Shortcuts</h2>
       <ul className="keyboard-shortcuts">{keyBoardShortcuts}</ul>
-      <p>
-        Note : You can use editor shortcuts using <kbd>f1</kbd>
-      </p>
+      <p>Note : You can use editor shortcuts using via clicking the 'Commands' button</p>
```
Comment (domoritz) [inline src/components/header/help-modal/index.tsx]:
```suggestion
      <p>Note: You can use editor shortcuts by clicking the 'Commands' button</p>
```

**Comment by domoritz** [comment]
Format still doesn't work correctly for me. I unchecked https://github.com/vega/editor/pull/379#issuecomment-509316344. 

Can you move the references above the shortcuts?

**Comment by algomaster99** [comment]
Screenshot from 2019-07-09 02-41-02

**Comment by algomaster99** [comment]
@domoritz 
Screenshot from 2019-07-09 13-50-05

**Comment by siddhant1** [comment]
This looks too good!

**Comment by algomaster99** [comment]
@domoritz 
> * Cannot run at all in automatic mode.
I am not sure if I get this. What is the problem exactly?

File: src/components/header/help-modal/index.tsx [hunk 2]
```diff
@@ -29,18 +29,23 @@ const HelpModal = () => {
...
-      </p>
+      <p>Note: You can use editor shortcuts by clicking the 'Commands' button</p>
       <a
         href={`https://github.com/vega/editor/issues/new?body=**Vega-Editor ${pjson.version}**`}
         target="_blank"
         className="report-button help-modal-link"
       >
         <AlertCircle className="header-icon" />
-        Report a Bug
+        <span>Report a Bug</span>
```
Comment (domoritz) [inline src/components/header/help-modal/index.tsx]: Why do you add a span?

File: src/components/input-panel/spec-editor/index.tsx [hunk 3]
```diff
@@ -22,6 +22,7 @@ export function mapDispatchToProps(dispatch: Dispatch<EditorActions.Action>) {
       formatSpec: EditorActions.formatSpec,
```
Comment (domoritz) [inline src/components/input-panel/spec-editor/index.tsx]: Can we remove this now that we have the commands?

File: src/components/header/help-modal/index.tsx (hunk 2 above)
Comment (algomaster99) [reply src/components/header/help-modal/index.tsx]: @domoritz I aligned the icon and the text using `display: flex`

File: src/components/input-panel/spec-editor/index.tsx (hunk 3 above)
Comment (algomaster99) [reply src/components/input-panel/spec-editor/index.tsx]: @domoritz I think we can now remove `this.props.format` and this function too.

File: src/components/header/help-modal/index.tsx (hunk 2 above)
Comment (domoritz) [reply src/components/header/help-modal/index.tsx]: Okay.

File: src/components/input-panel/spec-editor/index.tsx (hunk 3 above)
Comment (domoritz) [reply src/components/input-panel/spec-editor/index.tsx]: Go ahead.

**Pull request merged**